week3/tideman.c

```diff
@@ -0,0 +1,217 @@
+        }
+    }
+
+    pair_count = 0;
+    int voter_count = get_int("Number of voters: ");
+
+    for (int i = 0; i < voter_count; i++)
+    {
+        int ranks[candidate_count];
+
+        for (int j = 0; j < candidate_count; j++)
+        {
+            string name = get_string("Rank %i: ", j + 1);
+
+            if (!vote(j, name, ranks))
+            {
+                printf("Invalid vote.\n");
+
+                return 3;
+            }
+        }
+
+        record_preferences(ranks);
+        printf("\n");
+    }
+
+    add_pairs();
+    sort_pairs();
+    lock_pairs();
+    print_winner();
+
+    return 0;
+}
+
+bool vote(int rank, string name, int ranks[])
+{
+    for (int candidate_number = 0; candidate_number < candidate_count; candidate_number++)
+    {
+        if (strcmp(candidates[candidate_number], name) == 0)
+        {
+            ranks[rank] = candidate_number;
+
+            return true;
+        }
+    } 
+
+    return false;
+}
+
+void record_preferences(int ranks[])
+{
+    for (int rank = 0; rank < candidate_count; rank++)
+    {
+        for (int rank_after = rank+1; rank_after < candidate_count; rank_after++)
+        {
+            preferences[ranks[rank]][ranks[rank_after]]++;
+        }
+    }
+
+    return;
+}
+
+void add_pairs(void)
+{
+    for (int candidate_number = 0; candidate_number < candidate_count; candidate_number++)
+    {
+        for (int candidate_after = candidate_number+1; candidate_after < candidate_count; candidate_after++)
+        {
+            if (preferences[candidate_number][candidate_after] > preferences[candidate_after][candidate_number])
+            {
+                pairs[pair_count].winner = candidate_number;
+                pairs[pair_count].loser = candidate_after;
+                pair_count++;
+            }
+            else if (preferences[candidate_number][candidate_after] < preferences[candidate_after][candidate_number])
+            {
+                pairs[pair_count].winner = candidate_after;
+                pairs[pair_count].loser = candidate_number;
+                pair_count++;
+            }
+        }
+    }
+
+    return;
+}
+
+void sort_pairs(void)
+{
+    for (int pair_count_minus_one = (pair_count -1); pair_count_minus_one >= 0; pair_count_minus_one--)
+    {
+        for (int pair_number = 0; pair_number <= pair_count_minus_one; pair_number++)
+        {
+            if ((preferences[pairs[pair_number].winner][pairs[pair_number].loser]) < (preferences[pairs[pair_number+1].winner][pairs[pair_number+1].loser]))  
+            {
+                pair temporary = pairs[pair_number];
+                pairs[pair_number] = pairs[pair_number+1];
+                pairs[pair_number+1] = temporary;
+            }
+        }
+    }
+
+    return;
+}
+
+void lock_pairs(void)
+{
+    for (int pair_number = 0; pair_number < pair_count; pair_number++)
+    {
+        if(!has_cycle(pairs[pair_number].loser, pairs[pair_number].winner))
```

Comment:
```suggestion
        bool speakingvariable= ...;
        if(speakingVariable)
```
Hier würde eine Speaking variable gut passen. Also du verschiebst den boolean in einer vorher definierten Variable und packst die Speaking Variable in das If statement als Condition.